site/components/map/Map.tsx

```diff
@@ -261,14 +361,137 @@ const Map = ({
+            searchedRoutes.map((searchedRoute) =>
+                searchedRoute?.inbound ? (
+                    <Source
+                        key={searchedRoute?.serviceId}
+                        id={`inbound-route-${searchedRoute?.serviceId}`}
+                        type="geojson"
+                        data={createLineString(searchedRoute.inbound, searchedRoute?.serviceId)}
+                    >
+                        <Layer
+                            id={`services-inbound-${searchedRoute?.serviceId}`}
+                            type="line"
+                            source={`services-inbound-${searchedRoute?.serviceId}`}
+                            layout={lineLayout}
+                            paint={lineStyle}
+                        />
+                        <Layer
+                            id={`services-highlighted-inbound-${searchedRoute?.serviceId}`}
+                            type="line"
+                            source={`services-inbound-${searchedRoute?.serviceId}`}
+                            layout={lineLayout}
+                            paint={lineStyleHighlight}
+                            filter={filter}
+                        />
+                    </Source>
+                ) : null,
+            ),
+
+        [filter],
+    );
+
+    const getSourcesOutbound = useCallback(
+        (searchedRoutes: Partial<(Routes & { serviceId: number })[]>) =>
+            searchedRoutes.map((searchedRoute) =>
+                searchedRoute?.outbound ? (
+                    <Source
+                        key={searchedRoute?.serviceId}
+                        id={`outbound-route-${searchedRoute?.serviceId}`}
+                        type="geojson"
+                        data={createLineString(searchedRoute.outbound, searchedRoute?.serviceId)}
+                    >
+                        <Layer
+                            id={`services-outbound-${searchedRoute?.serviceId}`}
+                            type="line"
+                            source={`services-outbound-${searchedRoute?.serviceId}`}
+                            layout={lineLayout}
+                            paint={lineStyle}
+                        />
+                        <Layer
+                            id={`services-highlighted-outbound-${searchedRoute?.serviceId}`}
+                            type="line"
+                            source={`services-outbound-${searchedRoute?.serviceId}`}
+                            layout={lineLayout}
+                            paint={lineStyleHighlight}
+                            filter={filter}
+                        />
+                    </Source>
+                ) : null,
+            ),
+        [filter],
+    );
+
+    const getInteractiveLayerIds = useCallback(
+        () =>
+            searchedRoutes && searchedRoutes.length > 0
+                ? searchedRoutes.flatMap((sr) => {
+                      if (sr?.inbound && sr.outbound) {
```

Comment:
Do we need sr?.inbound instead of sr.inbound? Outbound seems ok without it. We can also check for serviceId so we dont have to do ?. below

Reply:
it complains about sr not being defined if i remove the sr?.

Reply:
Do we know why Inbound might be undefined on a searchRoute but oubound is never undefined?

Reply:
Or rather, do we know why it works for outbound and not for inbound

Reply:
done serviceId check

Reply:
sr.outbound only becomes defined when i check inbound first unless I do sr?.outbound it seems like if one exists the other also does

Reply:
Fair enough!